tests/addnode.py

```diff
@@ -12,6 +12,43 @@
 from loguru import logger as LOG
 
 
+def retire_node(primary, node_id):
```

Comment:
Would it make sense to move this function to `ccf.py` so that it can b easily reused?

Reply:
We could, but this might create some confusion when using the term "node_id".
The node id here is the one returned from the add_node API, and is not necessarily the node object's identifier. I will open a ticket for refactoring ccf.py so that each node's identifier will be similar to the one on the KV (maybe encapsulate the current id and add a "real_node_id" attribute?), since this problem keeps on reappearing.
I also think that the first thing that should be added is an API for voting, which can then be wrapped by function with specific proposal types.

Reply:
Awesome - let's do that in a different PR then.